 <launch>
  <!-- General arguments -->
  <arg name="param_file"
       default="$(find wavemap_ros)/config/alma_hashed_wavelet_adaptive.yaml"
       doc="Name of the file from which to read wavemap's server params."/>
  <arg name="debug_mode"
       default="false"
       doc="Run the mapper in a debugger (gdb)."/>
  <arg name="show_rviz"
       default="false"
       doc="Automatically launch Rviz."/>
  
<arg name="planner_frame"
     default="map"
     doc="Frame planner uses for planning (should match wavemap frame)."/>
<arg name="n_past_seconds"
     default="1"
     doc="Frame planner uses for planning (should match wavemap frame)."/>
  <arg name="load_map_from_file"
       default="false"
       doc="Load the map from a file."/>
  <arg name="param_file_waverider_planner"
      default="$(find waverider_chomp_planner)/config/waverider_planner_param.yaml"
      doc="Name of the file from which to read waverider planner's params."/>

  <!-- Run the wavemap server -->
  <!-- <include file="$(find wavemap_ros)/launch/wavemap_server.launch" unless="$(arg load_map_from_file)">
     <arg name="param_file" value="$(arg param_file)"/>
     <arg name="debug_mode" value="$(arg debug_mode)"/>
     <arg name="show_rviz" value="$(arg show_rviz)"/>
  </include> -->

  <!-- Run the waverider planner -->
  <node name="waverider_planner" pkg="waverider_chomp_planner" type="waverider_planner_node"
        output="screen" launch-prefix="">
    <param name="planner_frame" value="$(arg planner_frame)"/>
    <param name="n_past_seconds" value="$(arg n_past_seconds)"/>
    <param name="load_map_from_file" value="$(arg load_map_from_file)"/>
    <rosparam file="$(arg param_file_waverider_planner)"/>
  </node>

</launch>
